pkg/invariants/disruptionlegacyapiservers/invariant.go

```diff
@@ -144,6 +147,20 @@ func newDisruptionCheckerForOAuthCached(adminRESTConfig *rest.Config) (*disrupti
 
 func (w *availability) StartCollection(ctx context.Context, adminRESTConfig *rest.Config, recorder monitorapi.RecorderWriter) error {
 	var err error
+
+	coreClient, err := e2e.LoadClientset(true)
+	if err != nil {
+		return fmt.Errorf("unable to load client set: %v", err)
+	}
+	isMicroShift, err := exutil.IsMicroShiftCluster(coreClient)
+	if err != nil {
+		return fmt.Errorf("unable to determine if cluster is MicroShift: %v", err)
+	}
+	if isMicroShift {
+		w.notSupportedReason = "Platform MicroShift not supported"
```

Comment:
This looks overly broad.  You actually just want to avoid the checks for openshift and oauth apiservers, right?

Reply:
Done